<sdf version='1.6'>
  <world name='default'>
    <model name="kinect">
    <pose>0.5 0.5 0.036 0 0 0</pose>
    <static>true</static>
    <link name="link_kinect">
      <inertial>
        <mass>0.1</mass>
      </inertial>
      <collision name="collision">
        <geometry>
          <box>
            <size>0.073000 0.276000 0.072000</size>
          </box>
        </geometry>
      </collision>
      <visual name="visual">
        <geometry>
          <mesh>
            <uri>model://kinect/meshes/kinect.dae</uri>
          </mesh>
        </geometry>
      </visual>
      <sensor name="camera" type="depth">
        <update_rate>20</update_rate>
        <camera>
          <horizontal_fov>1.396263</horizontal_fov>
          <image>
            <width>640</width>
            <height>480</height>
            <format>B8G8R8</format>
          </image>
          <clip>
            <near>0.15</near>
            <far>3</far>
          </clip>
        </camera>
        <plugin name="camera_plugin" filename="libgazebo_ros_openni_kinect.so">
          <baseline>0.2</baseline>
          <alwaysOn>true</alwaysOn>
          <!-- Keep this zero, update_rate in the parent <sensor> tag
            will control the frame rate. -->
          <updateRate>0.0</updateRate>
          <cameraName>camera_ir</cameraName>
          <imageTopicName>/camera2/color/image_raw</imageTopicName>
          <cameraInfoTopicName>/camera2/color/camera_info</cameraInfoTopicName>
          <depthImageTopicName>/camera2/depth/image_raw</depthImageTopicName>
          <depthImageCameraInfoTopicName>/camera2/depth/camera_info</depthImageCameraInfoTopicName>
          <pointCloudTopicName>/camera2/depth/points</pointCloudTopicName>
          <frameName>camera2_link</frameName>
          <pointCloudCutoff>0.15</pointCloudCutoff>
          <pointCloudCutoffMax>3.0</pointCloudCutoffMax>
          <distortionK1>0</distortionK1>
          <distortionK2>0</distortionK2>
          <distortionK3>0</distortionK3>
          <distortionT1>0</distortionT1>
          <distortionT2>0</distortionT2>
          <CxPrime>0</CxPrime>
          <Cx>0</Cx>
          <Cy>0</Cy>
          <focalLength>0</focalLength>
          <hackBaseline>0</hackBaseline>
        </plugin>
      </sensor>
    </link>
  </model>
    <model name='ground_plane'>
      <static>1</static>
      <link name='link'>
        <collision name='collision'>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>400 400</size>
            </plane>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>100</mu>
                <mu2>50</mu2>
              </ode>
              <torsional>
                <ode/>
              </torsional>
            </friction>
            <contact>
              <ode/>
            </contact>
            <bounce/>
          </surface>
          <max_contacts>10</max_contacts>
        </collision>
        <visual name='visual'>
          <cast_shadows>0</cast_shadows>
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>400 400</size>
            </plane>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Grey</name>
            </script>
          </material>
        </visual>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
        <enable_wind>0</enable_wind>
      </link>
    </model>
    <gravity>0 0 -9.8</gravity>
    <magnetic_field>6e-06 2.3e-05 -4.2e-05</magnetic_field>
    <atmosphere type='adiabatic'/>
    <physics name='default_physics' default='0' type='ode'>
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>1</shadows>
    </scene>
    <spherical_coordinates>
      <surface_model>EARTH_WGS84</surface_model>
      <latitude_deg>0</latitude_deg>
      <longitude_deg>0</longitude_deg>
      <elevation>0</elevation>
      <heading_deg>0</heading_deg>
    </spherical_coordinates>
    <model name='Floor_room'>
      <link name='main'>
        <pose frame=''>0 0 0.01 0 -0 0</pose>
        <collision name='main_Collision'>
          <geometry>
            <box>
              <size>40 40 0.001</size>
            </box>
          </geometry>
          <max_contacts>10</max_contacts>
          <surface>
            <contact>
              <ode/>
            </contact>
            <bounce/>
            <friction>
              <torsional>
                <ode/>
              </torsional>
              <ode/>
            </friction>
          </surface>
        </collision>
        <visual name='main_Visual'>
          <geometry>
            <box>
              <size>40 40 0.001</size>
            </box>
          </geometry>
          <material>
            <script>
              <uri>model://Floor_room/materials/scripts</uri>
              <uri>model://Floor_room/materials/textures</uri>
              <name>floor_room</name>
            </script>
          </material>
        </visual>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
        <enable_wind>0</enable_wind>
      </link>
      <static>1</static>
      <pose frame=''>-0.008519 0.006897 0 0 -0 0</pose>
    </model>
    <state world_name='default'>
      <sim_time>0 0</sim_time>
      <real_time>200 256708750</real_time>
      <wall_time>1603829984 871223425</wall_time>
      <iterations>199182</iterations>
      <model name='Floor_room'>
        <pose frame=''>0 0 0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='main'>
          <pose frame=''>0 0 0.01 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model>
      <model name='capsicum_plant_3'>
        <pose frame=''>0.0 0.0 0.0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='link_capsicum_plant_3'>
          <pose frame=''>0.0 0.0 0.0 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model>
      <model name='capsicum_plant_4'>
        <pose frame=''>3.0 0.0 0.0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='link_capsicum_plant_4'>
          <pose frame=''>3.0 0.0 0.0 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model>
      <model name='capsicum_plant_5'>
        <pose frame=''>6.0 0.0 0.0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='link_capsicum_plant_5'>
          <pose frame=''>6.0 0.0 0.0 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model>
      <model name='capsicum_plant_6'>
        <pose frame=''>9.0 0.0 0.0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='link_capsicum_plant_6'>
          <pose frame=''>9.0 0.0 0.0 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model>
      <model name='capsicum_plant_7'>
        <pose frame=''>12.0 0.0 0.0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='link_capsicum_plant_7'>
          <pose frame=''>12.0 0.0 0.0 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model>
      <model name='capsicum_plant_8'>
        <pose frame=''>15.0 0.0 0.0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='link_capsicum_plant_8'>
          <pose frame=''>15.0 0.0 0.0 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model>
      
      <model name='ground_plane'>
        <pose frame=''>0 0 0 0 -0 0</pose>
        <scale>1 1 1</scale>
        <link name='link'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <velocity>0 0 0 0 -0 0</velocity>
          <acceleration>0 0 0 0 -0 0</acceleration>
          <wrench>0 0 0 0 -0 0</wrench>
        </link>
      </model>
      
    </state>
    <gui fullscreen='0'>
      <camera name='user_camera'>
        <pose frame=''>-3.65452 -0.882609 1.54108 0 0.228791 0.302828</pose>
        <view_controller>orbit</view_controller>
        <projection_type>perspective</projection_type>
      </camera>
    </gui>
    <model name='capsicum_plant_3'>
      <link name='link_capsicum_plant_3'>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://capsicum_plant_3/meshes/VG07_3.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <pose frame=''>0 0 0 0 -0 0</pose>
    </model>
    <model name='capsicum_plant_4'>
      <link name='link_capsicum_plant_4'>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://capsicum_plant_4/meshes/VG07_4.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <pose frame=''>0 0 0 0 -0 0</pose>
    </model>
    <model name='capsicum_plant_5'>
      <link name='link_capsicum_plant_5'>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://capsicum_plant_5/meshes/VG07_5.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <pose frame=''>0 0 0 0 -0 0</pose>
    </model>
    <model name='capsicum_plant_6'>
      <link name='link_capsicum_plant_6'>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://capsicum_plant_6/meshes/VG07_6.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <pose frame=''>0 0 0 0 -0 0</pose>
    </model>
    <model name='capsicum_plant_7'>
      <link name='link_capsicum_plant_7'>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://capsicum_plant_7/meshes/VG07_7.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <pose frame=''>0 0 0 0 -0 0</pose>
    </model>
    <model name='capsicum_plant_8'>
      <link name='link_capsicum_plant_8'>
        <visual name='visual'>
          <pose frame=''>0 0 0 0 -0 0</pose>
          <geometry>
            <mesh>
              <uri>model://capsicum_plant_8/meshes/VG07_8.dae</uri>
            </mesh>
          </geometry>
        </visual>
        <self_collide>0</self_collide>
        <enable_wind>0</enable_wind>
        <kinematic>0</kinematic>
      </link>
      <static>1</static>
      <pose frame=''>0 0 0 0 -0 0</pose>
    </model>
    
  </world>
</sdf>
